geode-lucene/src/upgradeTest/java/org/apache/geode/cache/lucene/LuceneSearchWithRollingUpgradeTestBase.java

```diff
@@ -225,12 +224,33 @@ protected static ClientCache createClientCache(Properties systemProperties,
 
 
   void putSerializableObjectAndVerifyLuceneQueryResult(VM putter, String regionName,
+      boolean luceneVersionMismatch,
       int expectedRegionSize, int start, int end, VM... vms) throws Exception {
     // do puts
     putSerializableObject(putter, regionName, start, end);
 
     // verify present in others
-    verifyLuceneQueryResultInEachVM(regionName, expectedRegionSize, vms);
+
+    if (!luceneVersionMismatch) {
+      verifyLuceneQueryResultInEachVM(regionName, expectedRegionSize, vms);
+    }
+  }
+
+  public boolean hasLuceneVersionMismatch(Host host) {
+    for (VM vm : host.getAllVMs()) {
+      int currentVersionOrdinal = 0;
+      for (KnownVersion productVersion : KnownVersion.getAllVersions()) {
+        if (vm.getVersion().equals(productVersion.getName())) {
+          currentVersionOrdinal = productVersion.ordinal();
+        } else if (vm.getVersion().equals(VersionManager.CURRENT_VERSION)) {
+          currentVersionOrdinal = KnownVersion.CURRENT_ORDINAL;
+        }
+      }
+      if (currentVersionOrdinal < KnownVersion.GEODE_1_15_0.ordinal()) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
This might be simpler as:
```
    return host.getAllVMs().get(0).invoke(() -> {
      List<InternalDistributedMember> allMembers = InternalDistributedSystem
          .getConnectedInstance()
          .getDistributionManager()
          .getViewMembers();

      for (InternalDistributedMember member : allMembers) {
        if (member.getVersionOrdinal() < KnownVersion.GEODE_1_15_0.ordinal()) {
          return true;
        }
      }
      return false;
    });
```

Reply:
The problem with this is on older versions before getDistributionManager is introduced, and it failed with NoSuchMethodError. For the newer version(after 1.5.0) it throws NullPointerException in the same place.

Reply:
I see, thanks for the explanation.